docs/source/usecases/VirtualFunctions.md

```diff
@@ -111,6 +111,61 @@ deviceInstance->setAField(someHostValue); // set some field on the host
 
 This is the solution that the code teams we have talked to have said is the most productive way to solve the problem.
 
+## But what if I do not really need the V-Tables on the device side?
+
+The performance critical part of a code might not use pointers to base class and dynamic polymorphism to compute a result from data.
+Thus, the device might not need to have a working virtual function table.
+Consider the following example:
+```c++
+struct Interface
+{
+    KOKKOS_DEFAULTED_FUNCTION
+    virtual ~Interface() = default;
+
+    KOKKOS_FUNCTION
+    virtual void operator()( const size_t) const = 0;
+};
+
+struct Implementation : public Interface
+{
+    KOKKOS_FUNCTION
+    void operator()(const size_t i) const override
+    { ... }
+
+    void apply(){
+        Kokkos::parallel_for("myLoop",10,
+            KOKKOS_CLASS_LAMBDA (const size_t i) { this->operator()(i); }
+        );}
+};
+
+int main ()
+{
+    ... 
+    auto implementationPtr_h = std::make_shared<Implementation>();
+    implementationPtr_h->apply();
+}
+```
+### What is the problem?
+
+Inside the `parallel_for` the `operator()` is called. As `Implementation` derives from the pure virtual class `Interface`, the 'operator()' is marked `override`.
+On ROCm 5.2 this results in a memory access violation.
+When executing the `this->operator()(i,v)` call, the runtime looks into the V-Table and dereferences a host pointer on the device.
+
+### But if that is the case, why does it work with NVCC?
+
+Notice, that the `parallel_for` is called from a pointer of type `Implementation` and not a pointer of type `Interface` pointing to an `Implementation` object.
+Thus, no V-Table lookup for the `operator()` would be necessary as it can be deduced from the context of the call that it will be the `operator()` in the class `Interface`.
```

Comment:
Mistake, I think, and a more technically explicit wording:
```suggestion
Thus, no V-Table lookup for the `operator()` would be necessary as it can be deduced from the context of the call that it will be `Implementation::operator()`.
```

Reply:
even better catch